x-pack/plugin/otel-data/src/main/resources/index-templates/metrics-service_destination.10m@template.yaml

```diff
@@ -0,0 +1,48 @@
+---
+version: ${xpack.oteldata.template.version}
+index_patterns: ["metrics-service_destination.10m.otel-*"]
+priority: 130
+data_stream:
+  hidden: true
+allow_auto_create: true
+_meta:
+  description: aggregated APM metrics template installed by x-pack
+  managed: true
+composed_of:
+  - metrics@tsdb-settings
+  - otel@mappings
+  - metrics-otel@mappings
+  - semconv-resource-to-ecs@mappings
+  - metrics@custom
+  - metrics-otel@custom
+ignore_missing_component_templates:
+  - metrics@custom
+  - metrics-otel@custom
+template:
+  settings:
+    index:
+      mode: time_series
+  mappings:
+    properties:
+      data_stream.type:
+        type: constant_keyword
+        value: metrics
+      metricset:
```

Comment:
Originally I wanted to define `constant_keyword` under `resource.attributes` for `metricset.interval` and under `attributes` for `metricset.name`, since that's where these are stored. But that'd again run into the usual issue of wiping out existing mapping for `resource.attributes`.

But as it seems, this works as well - this is also covered by tests.

Just calling out to make sure there is no unwanted implication I may missed.